### DISCUSSION TITLE
Last_called update delay

### DISCUSSION THREAD

**Discussion by notownblues** [alandtse/alexa_media_player#1485] [Q&A]
Hi there,

Thank you for this great integration. This ticket is more of a "feature request" than an issue, so please let me know if there's a more appropriate place to post this.

Basically I've noticed that the last_called_timestamp and last_called_summary usually take a few seconds to update whereas media_position_updated_at updates instantly.

The reason for this request is that I've been looking for ways to duck my AVR's volume on Alexa wake word, so Alexa can hear the command better, without all the TV background noise. I've been using the last_called_timestamp sensor in an automation to achieve this, but I've noticed that it can take up to 6-7 seconds to update sometimes which can be frustrating in this scenario. Is there any way to improve the latency and update that sensor quicker?

I tried to use the media_position_updated_at updates for a few days, but the sensor doesn't only update on wake word, but also on volume changes etc which can cause false triggers.

Thank you for your help!

**Comment by alandtse** [comment]
Last called will always be delayed since it requires a separate call and parsing of the activity log after a successful command is issued. You won't be able to adjust volume based on that because by the time we detect an activity, Alexa has already parsed it.

**Comment by notownblues** [comment]
Thanks @alandtse, makes sense.

Do you think there would be a way to have an "Wake Word" attribute that would be set to true or false when the wake word only is triggered, similar to the On-Device Activity node with Alexa Remote2 Node Red component (that also takes a few seconds to initiate unfortunately) but instantly? Or would the delay also apply like with the Last_called function? 

Thanks for your help 🙏🏻

**Reply by alandtse** [reply]
No. We have no idea that a command is initiated until a few seconds after when the activity log is updated.
